vortex-buffer/src/buffer_mut.rs

```diff
@@ -322,6 +323,34 @@ impl<T> BufferMut<T> {
         self.length += slice.len();
     }
 
+    /// Appends a slice of type `T`, growing the internal buffer as needed.
+    ///
+    /// # Example:
+    ///
+    /// ```
+    /// # use vortex_buffer::BufferMut;
+    ///
+    /// let mut builder = BufferMut::<u16>::with_capacity(10);
+    /// builder.extend_from_slice(&[42, 44, 46]);
+    ///
+    /// assert_eq!(builder.len(), 3);
+    /// ```
+    ///
+    /// # Safety
+    ///
+    /// Assumes the buffer has enough capacity to hold the slice
```

Comment:
This doc is in contradiction with the first line of the docstring.

And the example in the docstring is for the wrong function